pkg/logs/input/kubernetes/launcher.go

```diff
@@ -91,13 +104,51 @@ func (l *Launcher) run() {
 			l.addSource(service)
 		case service := <-l.removedServices:
 			l.removeSource(service)
+		case ops := <-l.retryOperations:
+			l.addSource(ops.service)
 		case <-l.stopped:
 			log.Info("Kubernetes launcher stopped")
 			return
 		}
 	}
 }
 
+func (l *Launcher) scheduleServiceForRetry(svc *service.Service) {
+	containerID := svc.GetEntityID()
+	ops, exists := l.pendingRetries[containerID]
+	if !exists {
+		b := &backoff.ExponentialBackOff{
+			InitialInterval:     500 * time.Millisecond,
+			RandomizationFactor: 0,
+			Multiplier:          2,
+			MaxInterval:         5 * time.Second,
+			MaxElapsedTime:      30 * time.Second,
+			Clock:               backoff.SystemClock,
+		}
+		b.Reset()
+		ops = &retryOps{
+			service:          svc,
+			backoff:          b,
+			removalScheduled: false,
+		}
+		l.pendingRetries[svc.GetEntityID()] = ops
```

Comment:
suggestion: I don't think there's much overhead here but I think this just reads better if we use:
```
l.pendingRetries[containerID] = ops
```